### PR TITLE
Add `data-hashtags` for Facebook and Twitter

### LINKED ISSUE #59: Support twitter and facebook hashtag
Please support twitter and facebook hashtag in the `<a>` tag.

### PR DESCRIPTION
**Pull request by BeEdward**
Hashtag and hastags are the most commonly used features on facebook and twitter respectively. Adding support to those two will make the library even more powerful.

This PR add two optional attribute `data-hashtag` for facebook and `data-hashtags` for twitter.

`data-hashtag` is a single key word without `#` while `data-hashtags` are comma seperated strings without `#`.

Closes #59

### CODE REVIEW

**Comment by kytta** [comment]
Hi, sorry for having kept you waiting. Let me review this...

File: src/shareon.js [hunk 1]
```diff
@@ -85,6 +85,12 @@ const init = () => {
               via: encodeURIComponent(
                 child.dataset.via || container.dataset.via || ""
               ),
+              hashtag: encodeURIComponent(
+                child.dataset.hashtag || container.dataset.hashtag || ""
+              ),
```
Comment (kytta) [inline src/shareon.js:90]: Let's delete this and only use `hashtags`

```suggestion
```

File: src/shareon.js [hunk 2]
```diff
@@ -85,6 +85,12 @@ const init = () => {
...
+              hashtags: encodeURIComponent(
+                child.dataset.hashtags?.replaceAll(', ', ',') || container.dataset.hashtags?.replaceAll(', ', ',') || ""
```
Comment (kytta) [inline src/shareon.js:92]: Not a fan of this extra data validation. To keep the bundle size small, let's not fix the website owners' errors

```suggestion
                child.dataset.hashtags || container.dataset.hashtags || ""
```

File: src/shareon.js [hunk 3]
```diff
@@ -17,7 +17,7 @@ import "./shareon.css";
  * }) => string}}
  */
 const urlBuilderMap = {
-  facebook: (d) => `https://www.facebook.com/sharer/sharer.php?u=${d.url}`,
+  facebook: (d) => `https://www.facebook.com/sharer/sharer.php?u=${d.url}${d.hashtag? `&hashtag=#${d.hashtag}` : ''}`,
```
Comment (kytta) [inline src/shareon.js:20]: Here, we'll take the first hashtag

```suggestion
  facebook: (d) => `https://www.facebook.com/sharer/sharer.php?u=${d.url}${d.hashtags? `&hashtag=#${d.hashtags.split(',')[0]}` : ''}`,
```

**Review by kytta — CHANGES_REQUESTED** [review]
Great job! Yet, I think two very similar properties will cause confusion. How about using just `hashtags` and then only taking the first one for Facebook? This, of course, will need to be reflected in the docs

**Comment by BeEdward** [comment]
> Great job! Yet, I think two very similar properties will cause confusion. How about using just `hashtags` and then only taking the first one for Facebook? This, of course, will need to be reflected in the docs

Sure. I can do that and will update shortly.

**Comment by kytta** [comment]
Looks good! I hope I'll be able to merge it after some manual testing before Monday.

**Review by kytta — APPROVED** [review]

**Comment by kytta** [comment]
Hey @BeEdward; can you elaborate on `hashtags` for the Facebook sharer? I have not found it in Facebook's documentation, and adding this parameter has no effect for me.

**Comment by BeEdward** [comment]
> Hey @BeEdward; can you elaborate on `hashtags` for the Facebook sharer? I have not found it in Facebook's documentation, and adding this parameter has no effect for me.

I couldn't find the original documentation page, but in there old api, do you offer a `hashtag` parameter you can specify in the url. Here is a stackoverflow link I found
https://stackoverflow.com/questions/30806280/sharing-link-with-a-preset-hashtag

In their new dialogue api, they still have a `hashtag` parameter
https://developers.facebook.com/docs/sharing/reference/share-dialog

And in this repo https://ellisonleao.github.io/sharer.js/ , the author provide a `hashtag` option for facebook that totally work.

**Comment by kytta** [comment]
> I couldn't find the original documentation page, but in there old api, do you offer a `hashtag` parameter you can specify in the url.
> In their new dialogue api, they still have a `hashtag` parameter 
> And in this repo , the author provide a `hashtag` option for facebook that totally work

Yup, thank you, I realized where the bug lies! The hash sign should be URL-encoded, or else it will parse as URL fragment

File: src/shareon.js [hunk 4]
```diff
@@ -17,7 +17,7 @@ import "./shareon.css";
  * }) => string}}
  */
 const urlBuilderMap = {
-  facebook: (d) => `https://www.facebook.com/sharer/sharer.php?u=${d.url}`,
+  facebook: (d) => `https://www.facebook.com/sharer/sharer.php?u=${d.url}${d.hashtags? `&hashtag=#${d.hashtags.split(',')[0]}` : ''}`,
```
Comment (kytta) [inline src/shareon.js:20]: For everything to work, we need to escape two things:

- the hash sign
- the comma

```suggestion
  facebook: (d) => `https://www.facebook.com/sharer/sharer.php?u=${d.url}${d.hashtags? `&hashtag=%23${d.hashtags.split('%2C')[0]}` : ''}`,
```

**Review by kytta — CHANGES_REQUESTED** [review]
There are two things left: the URL encoding of some characters (see line comment) and documentation. Could you append the README [here](https://github.com/diffus-me/shareon/blob/73a7467cb48484ff25c0a4a544738f2bc235edd2/README.md?plain=1#L137-L143) with something like:

> add `data-hashtags` to a Twitter or Facebook button to include a hashtags. The value should be a comma-separated list of words. Facebook supports only one hashtag

**Comment by BeEdward** [comment]
> There are two things left: the URL encoding of some characters (see line comment) and documentation. Could you append the README [here](https://github.com/diffus-me/shareon/blob/73a7467cb48484ff25c0a4a544738f2bc235edd2/README.md?plain=1#L137-L143) with something like:
> 
> > add `data-hashtags` to a Twitter or Facebook button to include a hashtags. The value should be a comma-separated list of words. Facebook supports only one hashtag

Sure. Will update momentarily.

**Comment by BeEdward** [comment]
@kytta I think this is ready for commit

**Comment by kytta** [comment]
Released in [v2.2.0](https://github.com/kytta/shareon/releases/tag/v2.2.0) 🎉

**Pull request merged**